pkg/controller/restore_controller.go

```diff
@@ -285,8 +286,38 @@ func (controller *restoreController) getValidationErrors(itm *api.Restore) []str
 	return validationErrors
 }
 
+func (controller *restoreController) fetchBackup(bucket, name string) (*api.Backup, error) {
+	backup, err := controller.backupLister.Backups(api.DefaultNamespace).Get(name)
+	if err == nil {
+		return backup, nil
+	}
+
+	if !apierrors.IsNotFound(err) {
+		return nil, err
+	}
+
+	glog.V(4).Infof("Backup %q not found in backupLister, checking object storage directly.", name)
+	var objErr error
+	backup, objErr = controller.backupService.GetBackup(bucket, name)
+	if objErr != nil {
+		glog.V(4).Infof("Backup %q not found in object storage.", name)
+		// return the original lister error
```

Comment:
Wouldn't it make more sense to return this error?

Reply:
yeah, i guess so - was originally thinking that in the case where the user just typo-ed the backup name, the first error might make more sense, but the second is actually fine. minor thing either way.